ui/app/models/totp.js

```diff
@@ -0,0 +1,173 @@
+/**
+ * Copyright (c) HashiCorp, Inc.
+ * SPDX-License-Identifier: BUSL-1.1
+ */
+
+import Model, { attr } from '@ember-data/model';
+import { alias } from '@ember/object/computed';
+import { computed } from '@ember/object';
+import { withModelValidations } from 'vault/decorators/model-validations';
+import { withExpandedAttributes } from 'vault/decorators/model-expanded-attributes';
+import lazyCapabilities, { apiPath } from 'vault/macros/lazy-capabilities';
+
+const ALGORITHMS = ['SHA1', 'SHA256', 'SHA512'];
+const DIGITS = [6, 8];
+const SKEW = [0, 1];
+
+const validations = {
+  account_name: [
+    { type: 'presence', message: "Account name can't be blank." },
+    {
+      type: 'containsWhiteSpace',
+      message:
+        "Account name contains whitespace. If this is desired, you'll need to encode it with %20 in API requests.",
+      level: 'warn',
+    },
+  ],
+  algorithm: [
+    {
+      validator: (value) => ALGORITHMS.includes(value),
+      message: 'Algorithm must be one of ' + ALGORITHMS.join(', '),
+    },
+  ],
+  digits: [
+    { validator: (value) => DIGITS.includes(value), message: 'Digits must be one of ' + DIGITS.join(', ') },
+  ],
+  period: [{ type: 'number', message: 'Period must be a number.' }],
+
+  skew: [{ validator: (value) => SKEW.includes(value), message: 'Skew must be one of ' + SKEW.join(', ') }],
+};
+
+@withModelValidations(validations)
+@withExpandedAttributes()
+export default class TotpModel extends Model {
+  @attr('string', {
+    readOnly: true,
+  })
+  backend;
+
+  @alias('account_name') name;
+
+  @attr('string', {
+    fieldValue: 'name',
+    editDisabled: true,
+  })
+  account_name;
+  @attr('string', {
+    editDisabled: true,
+    possibleValues: ALGORITHMS,
+    defaultValue: 'SHA1',
+  })
+  algorithm;
+  @attr('number', {
+    editDisabled: true,
+    possibleValues: DIGITS,
+    defaultValue: 6,
+  })
+  digits;
+  @attr('string', {
+    editDisabled: true,
+  })
+  issuer;
```

Comment:
small attr options blocks can be on a single line
```suggestion
  @attr('string', { editDisabled: true })
  issuer;
```